A. When make a hypothesis test, the comparison object needs enough data to make a meaningful comparison. For example, compare "Hillary vs. U.S. presidential election campaigns such as Trump cannot be used as the basis for hypothesis testing, because neither of them became president at the time of the campaign, and they could not get relevant data on their presidential performance.When there is not enough data, the future performance can also be speculated through the existing information. For example, assuming that Biden's presidential performance is to be predicted in some cases, the future situation can be speculated on the existing Biden's performance. The "null hypothesis" mentioned in the video should be more accurately expressed as "we cannot reject the original hypothesis", because the original hypothesis needs evidence to refute, not to prove that it is correct.
In addition, when it comes to "comparing the original hypothesis with the alternative hypothesis", it is more mathematically accurate to use the version of "less than or equal to", not just "less than".

1.An idea is testable if it can be measured or observed in a way that generates data, allowing for evidence-based conclusions. It needs to be clear, specific, and tied to something that can be empirically assessed.A good null hypothesis is clear, testable, and represents a statement of "no effect" or "no difference." And also it should be falsifiable.The null hypothesis suggests no effect or no difference, while the alternative hypothesis suggests that there is an effect or difference. The goal of hypothesis testing is to use data to reject or fail to reject the null hypothesis in favor of the alternative.

2.1)This is the value of each sample. If we have a sample, such as the height of a group of people, "xi" represents everyone's height data.
  2)This is the mean of the sample.It is the average of all sample data.
  3)This is the overall mean, that is, the real average of the whole group. This is usually the unknown value that we want to speculate through research or hypothesis testing.
  4)This is the total mean assumed in the original hypothesis. It is what we think of as the total mean, and we usually decide whether to reject this value through a hypothetical test.
  Sample statistics We get it from the sample and use it to estimate the overall parameters.By comparing the average of the sample and the assumed total mean to judge whether there is enough evidence to reject the original hypothesis.

3. We take into account the sampling distribution. The p value represents the probability of getting the current statistic or more extreme than it in this "original hypothesis is true" world. If the p value is very small, it means that it is very unusual to observe the current data when the original hypothesis is true, so we may reject the original hypothesis.

4.Take a example in the video, if the dog is innocent, by default, the presence of a garbage cap on the dog's head is the doing of someone else, but it is highly unlikely that a neighbor would climb into the house and put a garbage cap on the dog's head. But the reality is that the dog did have a garbage cover on his head, so the neighbor didn't do it, the dog did it himself

In [2]:
#5.
import numpy as np

n_simulations = 10000
n_couples = 124
observed_right_tilt = 80  
observed_proportion = observed_right_tilt / n_couples


simulated_right_tilts = np.random.binomial(n=n_couples, p=0.5, size=n_simulations)
simulated_proportions = simulated_right_tilts / n_couples


p_value = np.mean(simulated_proportions >= observed_proportion)


print(f"Observed proportion: {observed_proportion:.4f}")
print(f"Simulated p-value: {p_value:.4f}")


Observed proportion: 0.6452
Simulated p-value: 0.0007


As a result, there is a  strong evidence against the null hypothesis.It is believed that people do have a tendency to turn their heads to the right.

6. The value of p will not be represented by a fixed value. It will be represented by a range.No specific p-value can definitively prove the null hypothesis is true or false. It only tells us how strong the evidence is to either reject or fail to reject the null hypothesis.The smaller the value of p, the lower the likelihood of error

In [4]:
7.
import pandas as pd
import numpy as np

# Step 1: Create the patient data DataFrame
patient_data = pd.DataFrame({
    "PatientID": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    "Age": [45, 34, 29, 52, 37, 41, 33, 48, 26, 39],
    "Gender": ["M", "F", "M", "F", "M", "F", "M", "F", "M", "F"],
    "InitialHealthScore": [84, 78, 83, 81, 81, 80, 79, 85, 76, 83],
    "FinalHealthScore": [86, 86, 80, 86, 84, 86, 86, 82, 83, 84]
})

# Step 2: Calculate the health score change
patient_data['HealthScoreChange'] = patient_data['FinalHealthScore'] - patient_data['InitialHealthScore']

# Step 3: Define the hypothesis
# H0: Mean health score change <= 0 (no improvement)
# H1: Mean health score change > 0 (vaccine improves health scores)

# Step 4: Perform bootstrapping to simulate the null distribution
np.random.seed(42)  # For reproducibility
n_bootstrap = 10000  # Number of bootstrap samples
null_means = []

# Center the health score changes around zero (null hypothesis)
centered_health_score_changes = patient_data['HealthScoreChange'] - patient_data['HealthScoreChange'].mean()

# Generate bootstrap samples
for _ in range(n_bootstrap):
    bootstrap_sample = np.random.choice(centered_health_score_changes, size=len(centered_health_score_changes), replace=True)
    null_means.append(np.mean(bootstrap_sample))

# Step 5: Compute the observed test statistic (mean of health score changes)
observed_mean_change = patient_data['HealthScoreChange'].mean()

# Step 6: Calculate the one-sided p-value (proportion of null means >= observed mean)
p_value_one_tailed = np.mean(np.array(null_means) >= observed_mean_change)

# Output the result
print(f"Observed mean health score change: {observed_mean_change}")
print(f"One-sided p-value: {p_value_one_tailed}")


Observed mean health score change: 3.3
One-sided p-value: 0.0009


changes：
Calculating the mean change in health score as the observed test statistic.And focus on whether the health score improves (one direction) by comparing the observed statistic against the null distribution in only one direction. The p-value is computed as the proportion of the null distribution values that are greater than or equal to the observed mean change.

In a one-tailed test, the p-value is expected to be smaller compared to the two-tailed test because only looking at one direction. Therefore, it is easier to find evidence against the null hypothesis in a one-sided test.

Here’s a summary of our exchanges regarding your "Vaccine Data Analysis Assignment":

Initial Discussion: You mentioned working on a "Vaccine Data Analysis Assignment," which involves p-values and hypothesis testing. Specifically, you wanted to adjust the code from "Demo II of Week 5 TUT," which revisits "Question 8" from the Week 04 homework.
Request for Help with Code Adjustment: You requested guidance on modifying the code for a "one-sided" (or "one-tailed") hypothesis test instead of the "two-sided" version it originally provided. We discussed that the primary change would be in how the p-value is calculated—shifting from considering deviations in both directions (two-tailed) to focusing on one direction (one-tailed). This leads to a smaller p-value in a one-tailed test if the effect is in the hypothesized direction, as we are only considering one tail of the distribution.
Clarifying Hypothesis Test Interpretation: We also touched on how the interpretation of a one-sided test changes compared to a two-sided one, as a one-sided test only looks for evidence in one direction. Therefore, this test is more specific and may result in a smaller p-value when the direction aligns with the hypothesis.
If you need to expand or clarify any specific parts for your submission, feel free to ask!

https://chatgpt.com/share/67082011-8d74-8007-b7ce-ec08c1f951f9

8. Problem introduction：
   1）Describe how the experiment involving students tasting tea relates to the original Fisher’s tea experiment with Dr. Muriel Bristol.
   2）Null Hypothesis p=0.5, which states that students are guessing, and the probability of correctly identifying the pouring method is purely random, like flipping a coin.
     Alternative Hypothesis p>0.5, which suggests that students can reliably tell whether the tea or milk was poured first, with a probability of success greater than chance.
   Quantitative Analysis:

In [3]:
import numpy as np

# Set seed for reproducibility
np.random.seed(42)

# Simulation parameters
n_students = 80
correct_guesses = 49
p_null = 0.5  # Null hypothesis: probability of guessing correctly is 50%
simulations = 10000

# Simulate binomial experiment
simulated_data = np.random.binomial(n=n_students, p=p_null, size=simulations)

# Calculate p-value (proportion of simulations where correct guesses >= 49)
p_value = np.mean(simulated_data >= correct_guesses)
print("Simulated p-value:", p_value)

Simulated p-value: 0.0294


Findings and Discussion:
The p-value of 0.0294 is less than the typical significance level of 0.05. This means there is moderate evidence against the null hypothesis.

9. Yes